### DISCUSSION TITLE
Warning with tailwindcss classes

### DISCUSSION THREAD

**Discussion by brocard** [unovue/shadcn-vue#786] [General]
What about this warn `warn - The class data-[state=closed]:slide-out-to-top-[48%] is ambiguous and matches multiple utilities.`

**Comment by Saeid-Za** [comment]
What component are you getting this error at?
If this issue still persists, please create an issue.
Thanks!

**Reply by brocard** [reply]
`DialogContent` has this classes and `AlertDialogContent`

![Screenshot 2024-10-07 at 11 11 35](https://github.com/user-attachments/assets/72993533-0a00-4f84-ad75-29127ab303f0)

and the warning is something like this 

![Screenshot 2024-10-07 at 11 12 42](https://github.com/user-attachments/assets/bee4d0da-a83f-4a3f-905e-e9b62b99c4cb)

**Reply by Saeid-Za** [reply]
Would you please, provide your nuxt/tailwind config?

**Reply by brocard** [reply]
Yes sure this my tailwindcss config

```import` { kovayColors } from './app/themes/colors';

/** @type {import('tailwindcss').Config} */
export default {
  darkMode: ['class'],
  safelist: ['dark'],

  content: [
    './app/**/*.{js,vue,ts}',
    './components/**/*.{js,vue,ts}',
    './layouts/**/*.vue',
    './pages/**/*.vue',
    './plugins/**/*.{js,ts}',
    './app.vue',
    './error.vue',
  ],

  safelistColors: ['kovay', 'albescent'],

  theme: {
    container: {
      center: true,
      padding: '2rem',
      screens: {
        '2xl': '1400px',
      },
    },
    extend: {
      colors: {
        border: 'hsl(var(--border))',
        input: 'hsl(var(--input))',
        ring: 'hsl(var(--ring))',
        background: 'hsl(var(--background))',
        foreground: 'hsl(var(--foreground))',
        primary: kovayColors.primary,
        secondary: {
          DEFAULT: 'hsl(var(--secondary))',
          foreground: 'hsl(var(--secondary-foreground))',
        },
        destructive: {
          DEFAULT: 'hsl(var(--destructive))',
          foreground: 'hsl(var(--destructive-foreground))',
        },
        muted: {
          DEFAULT: 'hsl(var(--muted))',
          foreground: 'hsl(var(--muted-foreground))',
        },
        accent: {
          DEFAULT: 'hsl(var(--accent))',
          foreground: 'hsl(var(--accent-foreground))',
        },
        popover: {
          DEFAULT: 'hsl(var(--popover))',
          foreground: 'hsl(var(--popover-foreground))',
        },
        card: {
          DEFAULT: 'hsl(var(--card))',
          foreground: 'hsl(var(--card-foreground))',
        },
        kovay: kovayColors.primary,
        albescent: {
          50: '#fcf8f0',
          100: '#f8efdc',
          200: '#f3e4c9',
          300: '#e5c38c',
          400: '#daa25d',
          500: '#d18a3e', //d18a3e
          600: '#c37333',
          700: '#a25b2c',
          800: '#82492a',
          900: '#6a3e24',
          950: '#391e11',
        },
        midnight: '#121063',
        metal: '#565584',
        tahiti: '#3ab7bf',
        silver: '#ecebff',
        'bubble-gum': '#ff77e9',
        bermuda: '#78dcca',
      },
      borderRadius: {
        xl: 'calc(var(--radius) + 4px)',
        lg: 'var(--radius)',
        md: 'calc(var(--radius) - 2px)',
        sm: 'calc(var(--radius) - 4px)',
      },
      keyframes: {
        'accordion-down': {
          from: { height: 0 },
          to: { height: 'var(--radix-accordion-content-height)' },
        },
        'accordion-up': {
          from: { height: 'var(--radix-accordion-content-height)' },
          to: { height: 0 },
        },
        'collapsible-down': {
          from: { height: 0 },
          to: { height: 'var(--radix-collapsible-content-height)' },
        },
        'collapsible-up': {
          from: { height: 'var(--radix-collapsible-content-height)' },
          to: { height: 0 },
        },
      },
      animation: {
        'accordion-down': 'accordion-down 0.2s ease-out',
        'accordion-up': 'accordion-up 0.2s ease-out',
        'collapsible-down': 'collapsible-down 0.2s ease-in-out',
        'collapsible-up': 'collapsible-up 0.2s ease-in-out',
      },
    },
  },
  plugins: [require('tailwindcss-animate')],
};
```

**Reply by Saeid-Za** [reply]
This could be a result of a badly configured tailwind.
Could you please provide a minimal reproduction?

**Reply by brocard** [reply]
what exactly do you need?

**Reply by brocard** [reply]
This only show at console with `pnpm dev`

**Reply by Saeid-Za** [reply]
Hello There !
A minimal Repo where I could reproduce the issue on my system to further investigate the root problem.

**Reply by brocard** [reply]
Yes i can share a component (_the componte was generate with shadcn-vue cli_) and my tailwindcss config, its enough?

**Reply by Saeid-Za** [reply]
Sorry for late response, if you could provide them as a project in a public GitHub repository, that would be helpful, so that I could run the project and encounter the issue myself.

**Reply by brocard** [reply]
yep let me tray if i can push small project in public

**Reply by bensmaj** [reply]
I also get this warning. It appears to be coming from the DialogContent file

**Reply by razbakov** [reply]
Same for me. Any idea how to fix it?

**Comment by casatwy** [comment]
meet the same warning, but I don't know how to fix it

**Reply by casatwy** [reply]
problem solved!

in my original `tailwind.config.js`, the last line is
```typescript
plugins: [animate, typography, require("tailwindcss-animate")]
```

but the first line in `tailwind.config.js`, there already has
```typescript
const animate = require("tailwindcss-animate")
```

so, the `animate` is duplicated with `require("tailwindcss-animate")` in the last line, I removed `require("tailwindcss-animate")` in last line, it turns out to be fine, the warning disappears.
```typescript
plugins: [animate, typography]
```

**Reply by casatwy** [reply]
@vickon16 @dhannypramana

**Reply by vickon16** [reply]
Thank you @casatwy. I also had this duplicated plugin in my code. I guess it is because of the shadcn CLI overriding my tailwind.config.ts file

**Reply by AlexPaven** [reply]
Good find but I'm getting the warning and I went over the tailwind config and I don't have duplicate imports.

**Comment by vickon16** [comment]
I have the same warning from the dialog component from shadcn. How do we fix it?

**Comment by dhannypramana** [comment]
hi. im getting the same warning from AlertDialogContent, and DialogContent component.
any idea how to fix it?
thanks

**Reply by razbakov** [reply]
What's your tailwind config? Ensure animate isn't listed twice in the plugins.
